### ISSUE TITLE
Portuguese Translation

### ISSUE THREAD

**Issue by guntribam** [adonisjs/v5-docs#8]
# Translating process
## How to start?
First thing you should do is reply to this issue picking one of the following topics that have :zzz:. After that go to this [repo](https://github.com/guntribam/docs.adonisjs.com) and fork it. Create a branch with the name of the topic you pick and start translating each subsection.

## How do i merge my translation?
After you finish the translation of your topic, submit a PR on the branch `pt-translation-merging`. Other translator should check your translation for misspelling and translating conventions.

## Can i review other people PR?
Absolutely. Please look at our Translation conventions table in the end of this issue

## The topic that i choose is too big, how to get help?
It's essentialy the same process. Have someone fork your repo, create a branch with the subtopic translation, and submit to your topic branch as a PR. You can review his PR and get all the help you need. You can use this issue to call for help. 

## How do you control the stages of a Topic?
:zzz: No fork `:zzz:`
✍🏼 forked `:writing_hand:`
:eyes:  fork under review `:eyes:`
:green_book: translation merged `:green_book:`

---

# Guides

:green_book:  introduction-installation-release 
:writing_hand: Fundamentals [@guntribam/Fundamentals ](https://github.com/guntribam/docs.adonisjs.com/tree/Fundamentals)
:zzz: Http
:zzz: Views & Templates
:zzz: Validator
:zzz: Database
:zzz: ORM
:zzz: Authentication
:zzz: Security
:zzz: Digging deeper

# Reference 
## Database

:zzz: First Section(Connection, Connection manager, etc)
:zzz: ORM
:zzz: Relationships

## Validator

:zzz: Schema types
:zzz: Validation rules

## Views & Templates

:zzz: Globals
:zzz: Tags

# Cookbooks

:zzz: Using socket.io
:zzz: Writing tests
:zzz: Deployment
:zzz: Validator

# Releases

:zzz: Out of preview
:zzz: December release -2020
:zzz: October release -2020
:zzz: August release -2020
:zzz: Introducing AdonisJS V5 Preview

---

# Translation conventions
| English          | Portuguese                                                |
| -------------      | -------------                                                   |
| breaking changes | - mudanças não retrocompatíveis </br> - quebras de código |
| framework | framework |
| build | construção |
| release | - lançamento </br> - liberação  |

**Comment by guntribam** [comment]
Finished 
introduction-installation-release
Taking
Fundamentals

**Comment by gpbuil** [comment]
Hi Guys, I would like to contribute on this effort. How should I move forward?

**Comment by guntribam** [comment]
@gpbuil You can fork [my repo](https://github.com/guntribam/docs.adonisjs.com). Pick one topic like [http](https://docs.adonisjs.com/guides/context) for example, create a branch HTTP, and start translating each subtopic: Context, Routing and so on. After you finish submit a pull request to the branch pt-translation-merging. If someone want to help you in some subtopic they should fork your repo, and submit the translation to you as a PR into your branch HTTP for example...

If you decide to pick a topic please reply to this issue

**Comment by janainascal** [comment]
Hi people, I'd like do discuss the merge process. I imagine that at least one person should review and aprove before merging, to help check misspelling and other errors like that.

What do you thing?

**Comment by janainascal** [comment]
@guntribam I rewrited your list to  something more visual to help tracking the translation. 
If you like pelase edit your comment with this and add the link to your `Fundamentals` fork

Fork stages:
💤 No fork `:zzz:`
✍🏼 forked `:writing_hand:` 
👀  fork under review `:eyes:` 
📗 translation merged `:green_book:`

Other ideas for:
- forked: :book: :bookmark_tabs: :construction: :building_construction:   
- review: :mag: :mag_right: :passport_control:
- merged: :books: :ok: :heavy_check_mark: :white_check_mark: :ballot_box_with_check:

**Comment by janainascal** [comment]
> Hi Guys, I would like to contribute on this effort. How should I move forward?

You should follow [gunter reply](https://github.com/adonisjs/docs.adonisjs.com/issues/8#issuecomment-841553176)

There is also a portuguese community in adonisjs discord, in case you wanna ask for help or anything else 🙂

**Comment by guntribam** [comment]
@janainascal Thanks for the suggestions, I agreed with all. Some things to consider
- when posting a link to you forked repo i'll use this format for the link: @USERNAME/TOPIC it is ok?
- What do you think of the table of Translating conventions? When can use it in order to guide the review in some cases

**Comment by janainascal** [comment]
> * when posting a link to you forked repo i'll use this format for the link: @USERNAME/TOPIC it is ok?

It's ok for me 😄 

> * What do you think of the table of Translating conventions? When can use it in order to guide the review in some cases

Don't know, I never translate docs before 😅
But we can discuss this here and edit the first comment to track the covention current stage

**Comment by janainascal** [comment]
For example what do you thing about using `mundaça disruptiva`, `alteração significativa` or ` quebra de código` for `breaking change` 

| English | Portuguese | suggestions
-- | :-: | --
breaking changes | - | - mudanças não retrocompatíveis </br> - mundaças disruptivas  </br> - alterações significativas  </br> - quebras de código

For me `quebra de código` make more sense.

**Comment by guntribam** [comment]
If you google translate `breaking change` it suggests `mudança significativa` ou  `alteração significatica `. The problem with this kind of translation it is that doesn't reflect what is actually happening. `Meaningful change` it is not a `breaking change`. I'll keep `quebras de código` because that shows what is actually happening.

**Comment by guntribam** [comment]
@janainascal What do you think of these words:
- framework
- release
- build

I don't think we should translate them (i translate release for `lançamento` in my first branch, but i regret it). What do you think? We should put the on the Translation Convention table?

**Comment by janainascal** [comment]
<html>
<body>


English | Portuguese
-- | --
framework | - framework 
build | - construção
release | - versão estável <br/> - lançamento <br/> - liberação

</body>
</html>

Teoricamente só quem entende Português entrará aqui, então vou escrever em Português mesmo.

### framework

Se a [wikipédia](https://pt.wikipedia.org/wiki/Framework) não traduz eu também não hehe. Brincadeira.
Não achei nada utilizando uma versão traduzida dessa palavra e vi vários artigos mantendo-a sem traduzir. Acho que se tivesse uma palavra melhor para usar [essa pagina estável da wikipédia]((https://pt.wikipedia.org/wiki/Framework)) descreveria-a.

### build

Pelo _translate_ poderia ser compilação ou construção. Acho que pode ser construção mesmo. Não gosto de compilação pois não abrange tudo que o _build_ faz (estou me baseando nessa [resposta do stackoverflow](https://pt.stackoverflow.com/a/157143/148251))

### release
Essa palavra é mais complicada, pois ela depende muito do contexto em que está sendo usada, e algumas vezes até some da [tradução](https://www.linguee.com.br/ingles-portugues/traducao/software+release.html) 
<html>
<body>


This new software release became fully operational during the first quarter of 2010. | Este novo software tornou-se plenamente operacional no primeiro trimestre de 2010.
-- | --



</body>
</html>

A [definição que encontrei](https://www.bmc.com/blogs/software-deployment-vs-release/) se baseia no ISO/IEC - 20000-10 que diz:
> 3.2.13
release, noun
collection of one or more new or changed services (3.2.15) or service components (3.2.18) deployed into
the live environment as a result of one or more changes

Resumindo: é a disponibilização do código para o ambiente de _produção_. Se for um novo serviço, é o lançamento dele, então lançamento é uma tradução válida. Se for algo que já foi lançado, pode entender por **liberação** do código o ambiente de produção, ou uma atualização do produto.

Também encontrei o uso de palavra para se referir a uma versão do código, ou uma branch do git. Nesse contexto seria _release_ seria a versão estável, a qual eventualmente será **construída** e disponibilizada para o ambiente de produção.

**Comment by filipebraida** [comment]
Minha opnião que algumas palavras não deveriam ser traduzidas. Somente traduzidas se existir uma palavra já muito bem consolidada. Quando há dúvida na tradução, eu acho melhor repensar.  Acho que o caso do **build** é o mesmo do **release**. Dependendo do contexto, pode atrapalhar.

**Comment by janainascal** [comment]
> Minha opinião que algumas palavras não deveriam ser traduzidas. 

Concordo e estamos aqui para discutí-las.

> Somente traduzidas se existir uma palavra já muito bem consolidada. 

Já não concordo. Acho que mais importando é o entendimento do texto do que a preocupação da tradução de palavra por palavra.

> Acho que o caso do **build** é o mesmo do **release**. Dependendo do contexto, pode atrapalhar.

Você poderia dar um exemplo?

Acredito que devemos usar este issue para discutir exatamente este tipo de coisa, e se alguém mostrar erros no que usamos até o momento devemos avaliar e, dependendo, acrescentar/alterar a nossa lista de convenções

**Comment by jeanbispo** [comment]
Concordo com o @filipebraida palavras que rotineiramente são sempre usadas em inglês não deveriam ser traduzidas. Build, release, model, controller, são bons exemplos de palavras que fazem parte do dicionário diário do Dev e nunca ouvi alguém traduzindo na vida real.

**Comment by guntribam** [comment]
Eu estou de acordo com @filipebraida também, sou a  favor de não traduzir:
- framework
- build
- release

Como disse antes, apesar de ter traduzido `release` para lançamento na primeira seção, me arrependi e acho que é melhor manter Release.

Proponho de discutir esse tópico até no máximo quarta(19/05/2021). Para evitar a fadiga de discussões. Nessa data ponderamos a decisão da maioria e optamos por ela. Pode ser?

**Comment by janainascal** [comment]
Pode ser. Até não tenho nada contra em não traduzir palavras normalmente já usadas em inglês no nosso cotidiano.

Porém tenho 2 ressalvas:
1. Acredito que devamos traduzir ao máximo os textos para português DESDE QUE não atrapalhe o entendimento do texto.
    - ao máximo não significa tudo 😉  
2. Estejamos sempre abertos para discutir/analisar caso alguém discorde das nossas traduções. 
    - Isso **_não_** significa mudar o que já definimos para agradar a todos.

**Comment by filipebraida** [comment]
> Você poderia dar um exemplo?

Vou dar um exemplo de um caso que eu vi. No livro Linguagens de Programação do Sebesta, optaram por traduzir tudo. Traduziram _heap_ por monte. Eu já vi confundindo monte com: estrutura de dados, monte carlos, ... 

> Acredito que devamos traduzir ao máximo os textos para português DESDE QUE não atrapalhe o entendimento do texto.
Eu concordo, mas o ponto chave é esse limiar.

Algumas palavras não só possuem tradução específica e, sim, são quase um nome pessoal. Ela própria carrega mais do que o significado da tradução. Vejo que seria igual ao caso de traduzir um nome de uma pessoa para outra língua. Ao meu ver, ajuda os mais novos para a pesquisa. Uma pessoa entrando neste mundo, normalmente não sabe o que é _framework_. Imagina ela encontrando arcabouço no texto. 

> Proponho de discutir esse tópico até no máximo quarta(19/05/2021). Para evitar a fadiga de discussões. Nessa data ponderamos a decisão da maioria e optamos por ela. Pode ser?

Acho ótimo. Caso contrário, vai ficar eterno aqui.

> Build, release, model, controller, são bons exemplos de palavras que fazem parte do dicionário diário do Dev e nunca ouvi alguém traduzindo na vida real.

Essas palavras todas tem uma tradução "oficial". Algumas palavras como classe, associação, hierarquia, e outras, não vejo problema em traduzir. Nesses dois último casos que você deu (_model_ e _controller_) também não vejo muito problema traduzir. 

Não necessariamente as traduções oficiais vieram de pessoas da área. Acontece muito de outras pessoas traduzirem os textos técnicos e decidirem a tradução. Fica o caso do livro do Sebesta.

> Estejamos sempre abertos para discutir/analisar caso alguém discorde das nossas traduções. 

Sempre!

**Comment by janainascal** [comment]
>Essas palavras todas tem uma tradução "oficial".

Como assim "oficial"? É de uma fonte em que podemos nos basear?
Eu uso a Wikipédia em português, mas é uma _fonte_ e não a **regra**

Aproveitando o gancho, quais traduções vocês acham que seriam boas para _Model_, _View_ e _Controller_?
Na [wikipédia](https://pt.wikipedia.org/wiki/MVC) diz que é Modelo-Visão-Controle, mas encontrei traduções de _Controller_ para Controlador(a) e de _View_ para Visualização.   

Eu fico muito divida em traduzir ou não, pois acho _controlador(a)_ e _controle_ ambos ruins para nomenclatura de classe.
Por exemplo: 

UserController | ProfileController
-- | --
UsuárioControlador | PerfilControlador
ControladorUsuário | ControladorPerfil
ControladorDeUsuário | ControladorDePerfil
UsuárioControle | PerfilControle
ControleUsuário | ControlePerfil
ControleDeUsuário | ControleDePerfil

Já _Modelo_ funciona bem: ModeloUsuário, UsuárioModelo, ModeloDeUsuário

**Comment by filipebraida** [comment]
Sim, concordo. Wikipedia não é fonte e sim livros. Falei mais que já vi em livros algumas traduções, como exemplo arcabouço.

Eu fico dividido em traduzir código. Não pelo nome e sim pela rastreabilidade. Não tenho algo muito bem formado sobre isso. Tem seu lado bom e ruim.

**Comment by janainascal** [comment]
Voltando ao _release_ e _build_, acho que como verbos deveríamos traduzir mas como substantivo não
Inglês | Português
-- | --
the build folder | a pasta _build_
to build the project | Para construir o projeto

**Comment by janainascal** [comment]
Até eu começar nesse meu emprego atual eu também era contra, mas como era só eu que escrevia em inglês e todo o projeto legado estava em português, escrever em inglês atrapalhava a minha comunicação com o resto do time e eu levava muito tempo para decidir a nomenclatura.

 Por exemplo, uma coluna de uma tabela chama-se `bom_para`, é a data na qual a pessoa que gerou a nota, ou o cheque, terá dinheiro na conta para que a pessoa que recebeu possa sacar. O famoso cheque pré-datado.
 
 Em português já é difícil de mudar, imagina em inglês.
 
 Além disso, se formos seguir o **DDD**, a nomenclatura a ser implantada deve seguir o domínio, então eu não deveria renomear a coluna e deviria usar esse nome `bomPara` no meu modelo de domínio. 
 
 Assim sendo não estou traduzindo código para português, estou criando código em português.

**Comment by janainascal** [comment]
De volta ao _release_

Acredito que essa palavra é uma redução de _release version_ e por isso é tão estranho para nós traduzirmos, pois traduzimos para lançamento/liberação quando deveria ser algo tipo versão lançada/liberada, ou lançamento/liberação de versão.

**Comment by filipebraida** [comment]
> Até eu começar nesse meu emprego atual eu também era contra, mas como era só eu que escrevia em inglês e todo o projeto legado estava em português, escrever em inglês atrapalhava a minha comunicação com o resto do time e eu levava muito tempo para decidir a nomenclatura.

É, nesse caso é ruim mesmo.

>  Acredito que essa palavra é uma redução de release version e por isso é tão estranho para nós traduzirmos, pois traduzimos para lançamento/liberação quando deveria ser algo tipo versão lançada/liberada, ou lançamento/liberação de versão.

Poderia ser lançamento de versão.

> Eu fico muito divida em traduzir ou não, pois acho controlador(a) e controle ambos ruins para nomenclatura de classe.

Eu fico na dúvida por causa dos geradores de código. Ele vai manter o padrão e, talvez, fique confuso para o leitor. No texto poderia falar de controlador, modelo, ... Eu fiquei refletindo sobre o código e pensei nessa questão.

**Comment by janainascal** [comment]
Creio que o melhor seria traduzir no texto mas manter no código. Por exemplo, [nessa seção](https://docs.adonisjs.com/guides/controllers) poderia ser algo tipo:

>In AdonisJS, the controllers are stored inside (but not limited to) the `app/Controllers/Http` directory and each file represents a single controller. 

> No AdonisJS, os controladores são armazenados dentro da (mas não limitado a) pasta `app/Controllers/Http` e cada arquivo representa um controlador

```javascript
// app/Controllers/Http/PostsController.ts

export default class PostsController {
...
}
```

**Comment by dipedro** [comment]
> Voltando ao _release_ e _build_, acho que como verbos deveríamos traduzir mas como substantivo não

Essa abordagem, pra mim, é a mais sensata e correta!
Inclusive resolve o problema do release, build, framework, controller, model, view e de outras palavras reservadas da linguagem/tecnologia.
